### PR TITLE
[BPK-2257] Align flow version to react native's version

### PR DESCRIPTION
**Pull request by matthewdavidson**
Also adopt the flowconfig that ships with a newly created RN app

### CODE REVIEW

File: packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js [hunk 1]
```diff
@@ -63,6 +63,7 @@ const BpkBadgeIcons = (props: Props) => {
         <Fragment key={`${icon.props.icon}`}>
           {icon}
           {separator && index < adjustedIcons.length - 1 && (
+            // $FlowFixMe
```
Comment (k0nserv) [inline packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js]: Can we include a comment with more details when we use `$FlowFixMe`?

Comment (matthewdavidson) [reply packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js]: Yeah of course, I just have to remember why ... 😂

Comment (shaundon) [reply packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js]: I removed the `// $FlowFixMe` so I could see what the error was, and I'm none the wiser..

screen shot 2019-02-21 at 11 29 08

File: packages/react-native-bpk-component-calendar/src/BpkCalendar.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ export type Props = {
 };
 
 const createOnChangeHandler = memoize(callback => event => {
+  // $FlowFixMe
```
Comment (shaundon) [inline packages/react-native-bpk-component-calendar/src/BpkCalendar.js]: I don't think you need this, I removed it and 'sall good man

File: packages/react-native-bpk-component-carousel/src/BpkCarousel-test.common.js [hunk 3]
```diff
@@ -42,7 +42,11 @@ const renderWithLayout = (
       layout,
     },
   };
-  rendered.toJSON().props.onLayout(mockNativeEvent);
+  const json = rendered.toJSON();
+  if (!json) {
+    throw new Error('');
```
Comment (shaundon) [inline packages/react-native-bpk-component-carousel/src/BpkCarousel-test.common.js]: Can we make this error more descriptive?

Comment (matthewdavidson) [reply packages/react-native-bpk-component-carousel/src/BpkCarousel-test.common.js]: oops

File: packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js (hunk 1 above)
Comment (matthewdavidson) [reply packages/react-native-bpk-component-badge/src/BpkBadgeIcons.js]: If I remove `StyleSheet.flatten` from the styles being applied to that `BpkText` component then it goes away 🔮

File: packages/react-native-bpk-component-image/README.md [hunk 4]
```diff
@@ -72,7 +72,6 @@ export default () => (
 
 | Property        | PropType                               | Required  | Default Value   |
 | --------------- | -------------------------------------- | --------- | --------------- |
-| alt             | string                                 | true      | -               |
```
Comment (shaundon) [inline packages/react-native-bpk-component-image/README.md]: How come this is gone?

File: packages/react-native-bpk-component-text-input/src/BpkTextInput.js [hunk 5]
```diff
@@ -20,14 +20,9 @@
...
-import {
-  Animated,
-  type AnimatedValue,
-  TextInput,
-  View,
-  ViewPropTypes,
-} from 'react-native';
+import { Animated, TextInput, View, ViewPropTypes } from 'react-native';
 import BpkText from 'react-native-bpk-component-text';
+import AnimatedValue from 'react-native/Libraries/Animated/src/nodes/AnimatedValue';
```
Comment (shaundon) [inline packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: It's kind of sucky that we need to do this 😭

File: packages/react-native-bpk-component-text-input/src/styles.js [hunk 6]
```diff
@@ -17,7 +17,8 @@
  */
 /* @flow */
 
-import { StyleSheet, I18nManager, type AnimatedValue } from 'react-native';
+import { StyleSheet, I18nManager } from 'react-native';
+import AnimatedValue from 'react-native/Libraries/Animated/src/nodes/AnimatedValue';
```
Comment (shaundon) [inline packages/react-native-bpk-component-text-input/src/styles.js]: Do you think there's any merit to us abstracting this and some other common types out somewhere, so we could then do something like:

```
import { AnimatedValue, otherStuff } from 'some-convenient-package';
```

I'm concerned that these paths are hard to find, e.g. `'react-native/Libraries/Animated/src/nodes/AnimatedValue'` and there's nothing to stop them changing in RN. If we abstract them and it does change, we can update it in one place.

File: packages/react-native-bpk-component-image/README.md (hunk 4 above)
Comment (matthewdavidson) [reply packages/react-native-bpk-component-image/README.md]: It wasn't being used ... at all!

File: packages/react-native-bpk-component-text-input/src/styles.js (hunk 6 above)
Comment (matthewdavidson) [reply packages/react-native-bpk-component-text-input/src/styles.js]: We can create our own libdef for it - but I reckon that's just putting a layer of indirection that would potentially mask errors if underlying types in react native change.

As for minimising the number of files we'd need to change if the file changes path or what not ... i'm a little indifferent tbh. Nothing a search + replace cant fix?

Comment (shaundon) [reply packages/react-native-bpk-component-text-input/src/styles.js]: Yeah fair enough - perfect being the enemy of good and all, let's just leave it as is.

File: packages/react-native-bpk-component-image/README.md (hunk 4 above)
Comment (shaundon) [reply packages/react-native-bpk-component-image/README.md]: Oh haha

**Comment by matthewdavidson** [comment]
@k0nserv can you approve please?

**Pull request merged**